applications/mne_scan/plugins/eegosports/eegosportsdriver.cpp

```diff
@@ -91,42 +96,94 @@ EEGoSportsDriver::~EEGoSportsDriver()
 
 //*************************************************************************************************************
 
-bool EEGoSportsDriver::initDevice(int iNumberOfChannels,
-                            int iSamplesPerBlock,
-                            int iSamplingFrequency,
-                            bool bWriteDriverDebugToFile,
-                            QString sOutpuFilePath,
-                            bool bMeasureImpedance)
+bool EEGoSportsDriver::initDevice(bool bWriteDriverDebugToFile,
+                            QString sOutpuFilePath, bool bMeasureImpedance)
 {
+    m_bMeasureImpedances = bMeasureImpedance;
+
     //Check if the driver DLL was loaded
     if(!m_bDllLoaded) {
         return false;
     }
 
     //Set global variables
-    m_uiNumberOfChannels = iNumberOfChannels;
-    m_uiSamplesPerBlock = iSamplesPerBlock;
-    m_uiSamplingFrequency = iSamplingFrequency;
     m_bWriteDriverDebugToFile = bWriteDriverDebugToFile;
     m_sOutputFilePath = sOutpuFilePath;
-    m_bMeasureImpedances = bMeasureImpedance;
 
     //Open debug file to write to
     if(m_bWriteDriverDebugToFile) {
-        m_outputFileStream.open("./resources/mne_scan/plugins/eegosports/EEGoSports_Driver_Debug.txt", std::ios::trunc); //ios::trunc deletes old file data
+        m_outputFileStream.open("./EEGoSports_Driver_Debug.txt", std::ios::trunc); //ios::trunc deletes old file data
     }
 
     try {
         // Get device handler
+#ifdef _WIN32
         factory factoryObj ("eego-SDK.dll"); // Make sure that eego-SDK.dll resides in the working directory
+#else
+        factory factoryObj ("/home/vorwerkj/git/mne-cpp/lib/libeego-SDK.so"); // Make sure that eego-SDK.dll resides in the working directory
+#endif
+
         m_pAmplifier = factoryObj.getAmplifier(); // Get an amplifier
 
         //std::cout<<"EEGoSportsDriver::initDevice - Serial number of connected eegosports device: "<<m_pAmplifier->getSerialNumber()<<std::endl;
 
+#ifdef _WIN32
+        Sleep(100);
```

Comment:
how about [QThread's sleep](https://doc.qt.io/qt-5/qthread.html#msleep)

Reply:
Sure, wasn't aware of that. Only wanted to make it work on mac/linux, as windows-only Sleep was originally used.